pkg/pillar/cmd/logmanager/logmanager.go

```diff
@@ -525,48 +577,107 @@ func processEvents(image string, logChan <-chan logEntry,
 			log.Debugf("processEvents(%s): sending at messageCount %d, byteCount %d\n",
 				image, messageCount, byteCount)
 			if event.isAppLog {
-				sent = sendProtoStrForAppLogs(appUUID, appLogBundle, iteration, image)
+				sent, is4xx = sendProtoStrForAppLogs(appUUID, appLogBundle, iteration, image)
+				if is4xx {
+					logMetrics.Num4xxResponses += uint64(messageCount)
+				} else {
+					logMetrics.NumAppEventsSent += uint64(messageCount)
+					logMetrics.NumAppBundleProtoBytesSent += uint64(byteCount)
+					logMetrics.NumAppBundlesSent++
+					logMetrics.LastAppBundleSendTime = time.Now()
+				}
 				delete(appLogBundles, appUUID)
 			} else {
-				sent = sendProtoStrForLogs(reportLogs, image,
-					iteration, eveVersion)
+				sent = sendProtoStrForLogs(reportLogs, image, iteration, eveVersion)
+				logMetrics.NumDeviceEventsSent += uint64(messageCount)
+				logMetrics.NumDeviceBundleProtoBytesSent += uint64(byteCount)
+				logMetrics.NumDeviceBundlesSent++
+				logMetrics.LastDeviceBundleSendTime = time.Now()
```

Comment:
Makes sense to document (In the API?) that "sent" really means that we tried to send, since you are increasing this even if "sent" is false i.e., things were deferred.

Alternatively we need a hook in the defer processing to be able to increase the counters then.

Reply:
To me it makes sense to document API saying that deferred logs are also counted as sent. Because, these logs will eventually reach cloud. Will add comment in metrics.proto

Reply:
done